app/src/androidTest/java/org/wikipedia/pages/SearchBarPage.kt

```diff
@@ -0,0 +1,58 @@
+package org.wikipedia.pages
+
+import androidx.test.espresso.Espresso.onView
+import androidx.test.espresso.action.ViewActions.*
+import androidx.test.espresso.matcher.ViewMatchers.*
+import org.hamcrest.Matchers.allOf
+import org.wikipedia.R
+
+
+
+class SearchBarPage : BasePage() {
+
+    private var text: String = ""
+    private val searchTextInput = withId(R.id.search_src_text)
+    private val pageList = withId(R.id.page_list_item_title)
+    private val recentSearchesDeleteButton = withId(R.id.recent_searches_delete_button)
+    private val yesButton = withText(R.string.dialog_message_clear_history_yes)
+    private val emptyImage = withId(R.id.search_empty_image)
+    private val resultText = hasDescendant(withText(text))
+
+
+    fun typeTextSearch(text: String) {
+        onView(searchTextInput).perform(typeText(text))
+    }
+
+    fun isDisplayedElement(): Boolean {
+        return viewExists(resultText)
+    }
+
+    fun isDisplayedNoResults():Boolean {
+        return waitForView(hasDescendant(withText("No results")))
```

Comment:
Move the `hasDescendant(withText("No results"))` to a new matcher variable on top of the file, together with other view matches, and use it here, in similar way like you did with `typeTextsSearch()` method

Reply:
I changed the matcher for this method